packages/runtime-vapor/src/component.ts

```diff
@@ -49,11 +69,22 @@ export const createComponentInstance = (
     block: null,
     container: null!, // set on mount
     scope: new EffectScope(true /* detached */)!,
-
     component,
-    isMounted: false,
+
+    // resolved props and emits options
+    propsOptions: normalizePropsOptions(component),
+    // emitsOptions: normalizeEmitsOptions(type, appContext), // TODO:
+
+    proxy: null,
+
+    // state
+    props: EMPTY_OBJ,
+    setupState: EMPTY_OBJ,
 
     dirs: new Map(),
+
+    // lifecycle hooks
+    isMounted: false,
```

Comment:
Note that in vapor we have no repeat-executing render function, so we have to mark the necessary state with reactive. In the latest commit, I added `isMountedRef` and `isMounted` is a proxied-getter for `isMountedRef`.

Please merge/rebase your PR to the main branch.